lib/ledgerBlockStorage.js

```diff
@@ -26,55 +26,58 @@ module.exports = class LedgerBlockStorage {
 
   /**
    * Adds a block in the ledger given a block, metadata associated with the
-   * block, and a set of options.
+   * block
    *
    * @param block - the block to create in the ledger.
    * @param meta - the metadata associated with the block.
    *   blockHash - the hash value of the block.
-   * @param options - a set of options used when creating the block.
    * @param callback(err) - the callback to call when finished.
    *   err - An Error if an error occurred, null otherwise.
    *   result - the result of the operation.
    *     block - the block that was committed to storage.
    *     meta - the metadata that was committed to storage.
    */
-  add(block, meta, options, callback) {
-    if(typeof options === 'function') {
-      callback = options;
-      options = {};
-    }
+  add({block, meta}, callback) {
     if(!meta.blockHash) {
       return callback(new BedrockError(
         'A block hash for the given block was not specified.',
-        'BadRequest', {meta}));
+        'DataError', {meta}));
     }
-
+    const {blockHeight, event} = block;
+    // event does not get saved as part of the block
+    delete block.event;
     async.auto({
       insert: callback => {
         // insert the block
         const now = Date.now();
         const record = {
-          id: database.hash(block.id),
+          block,
           blockHash: database.hash(meta.blockHash),
-          block: block,
+          id: database.hash(block.id),
           meta: _.defaults(meta, {
             created: now,
             updated: now
-          })
+          }),
         };
 
-        logger.debug('adding block: ' + meta.blockHash);
+        logger.debug(`adding block: ${meta.blockHash}`);
 
         // FIXME: We should deconstruct the events from the blocks
 
-        this.collection.insert(
-          record, database.writeOptions, (err, result) => {
-            if(err) {
-              return callback(err);
-            }
-            callback(null, result.ops[0]);
-          });
-      }
+        this.collection.insert(record, database.writeOptions, (err, result) => {
+          if(err) {
+            return callback(err);
+          }
+          callback(null, result.ops[0]);
+        });
+      },
+      update: ['insert', (results, callback) => {
+
+        const query = {eventHash: {$in: event.map(h => database.hash(h))}};
+        this.eventCollection.update(
+          query, {$set: {'meta.blockHeight': blockHeight}},
+          database.writeOptions, callback);
```

Comment:
What happens if we crash in the middle of this call? How do we recover and make sure all of the events are updated? Does the consensus algorithm handle that properly? I think the current code (at least in continuity) assumes writing the block always works ... so we'd need to be able to "finish adding" a block or something.